libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid-wrapper.component.ts

```diff
@@ -171,38 +186,23 @@ export class SkyAgGridWrapperComponent
   }
 
   public ngOnInit(): void {
-    this.#themeSvc?.settingsChange
+    combineLatest<[boolean, SkyThemeSettingsChange | undefined]>([
+      this.#hasEditableClass.pipe(distinctUntilChanged()),
+      this.#themeSvc?.settingsChange ?? of(undefined),
+    ])
       .pipe(takeUntil(this.#ngUnsubscribe))
-      .subscribe((settings) => {
-        this.#updateGridTheme(settings.currentSettings);
-        if (!this.#currentTheme) {
-          // Initial theme settings.
-          this.#currentTheme = settings.currentSettings;
-        } else if (this.agGrid?.api) {
-          // On subsequent theme changes, we need to call the api to re-render the grid.
-          this.#currentTheme = settings.currentSettings;
-          this.agGrid.api.setHeaderHeight(
-            this.#agGridService.getHeaderHeight(),
-          );
-          this.agGrid.api.resetRowHeights();
-          this.agGrid.api.refreshCells();
-        }
+      .subscribe(([hasEditableClass, settings]) => {
+        this.#updateGridTheme(hasEditableClass, settings?.currentSettings);
```

Comment:
so it looks like we update the theme if editable or the theme changes, and when we update theme we check if it is compact. Does it trigger an update when the compact input changes, or would that change not take effect until the editable state or themeSettings changed? I could see that being something that consumers give the user the option to toggle between?

Reply:
Good call. Updated.